types/react/index.d.ts

```diff
@@ -348,7 +348,7 @@ declare namespace React {
     }
 
     interface RefForwardingComponent<T, P = {}> {
-        (props: P & { children?: ReactNode }, ref?: Ref<T>): ReactElement<any> | null;
+        (props: P & { children?: ReactNode }, ref: Ref<T>): ReactElement<any> | null;
```

Comment:
They _are_ allowed to be `undefined`, though.

By the way, we probably should revise the type of `Ref<T>`, because `RefForwardingComponent` cannot receive string refs. It has been deprecated for a few years now, and `createRef` did create a smoother migration path... maybe add a `LegacyRef<T> = Ref<T> | string` for the components that do work with string refs?

Reply:
`undefined` from whose perspective?

Reply:
```js
const Forwarder = React.forwardRef(
  (props, ref) => <div {...props} ref={ref}/>
);
<Forwarder/>; // ref is undefined both here and inside the ref forwarder
```